### DISCUSSION TITLE
JReleaser interfaces with Installers

### DISCUSSION THREAD

**Discussion by ebresie** [jreleaser/jreleaser#793] [Q&A]
I'm very new to jreleaser but does jreleaser  interfaces with installers in any way to install on different OS platforms?

And does it handle all the related artifact (top level or embedded) signing activities?

**Comment by aalmiray** [comment]
Short answer for both questions is: **yes** 😄 

One of JReleaser's goals is to streamline delivery of binaries to end consumers, this is done by generating the set of required files by a given package manager. JReleaser currently supports the following packagers, many of which are cross-platform:

https://jreleaser.org/guide/latest/integrations.html#_package_publish

JReleaser expects binaries to be created by your build tool of choice however it also supports assembling some of those binaries using: [GraalVM Native Image](https://jreleaser.org/guide/latest/configuration/assemble/native-image.html), [Jlink](https://jreleaser.org/guide/latest/configuration/assemble/jlink.html), and [JPackage](https://jreleaser.org/guide/latest/configuration/assemble/jpackage.html). JPackage may be used to create platform installers (DMG, PKG, RPM, DEB, MSI) that in turn also be distributed/released with JReleaser.

Now, regarding signatures, the answer is not a strict **yes** but more of a **it depends**. JReleaser can [sign](https://jreleaser.org/guide/latest/configuration/signing.html) binaries with either PGP or Sigstore's cosign. However it **does not** sign binaries with Microsoft's signtool ([#689](https://github.com/jreleaser/jreleaser/issues/689)), nor Apple's codesign, nor notarizes/staples binaries with Apple's App Store ([#690](https://github.com/jreleaser/jreleaser/issues/690)). Also, container images created by the [Docker](https://jreleaser.org/guide/latest/configuration/packagers/docker.html) packager are not yet signed ([#688](https://github.com/jreleaser/jreleaser/issues/688)).
